src/Model/mutators.ts

```diff
@@ -10,6 +10,8 @@ var RemoveEvent = mutationEvents.RemoveEvent;
 var MoveEvent = mutationEvents.MoveEvent;
 var promisify = util.promisify;
 
+type ValueCallback<T> = ((error: Error | null | undefined, value?: T) => void);
```

Comment:
I dont understand the history of cb but usually we want to name that cb as to the event that the callback would fire.. for example all of these look like onCompleted callbacks where the caller will look to the props to see if there was an error to handle it. .. I would explicity type all the callbacks the same so type CompletedCallback where in some cases there maybe a value (in the case of a successful add there is a value.) 

OR 
i would explicity create callback events for the events themselves but that would require more refactoring onError onSuccess in the case of your ID i'm assuming its a success case?

Reply:
This explains the conventions behind "Node-style" (error-first) callbacks:
https://gist.github.com/sunnycmf/b2ad4f80a3b627f04ff2

One benefit of that style is that it makes it obvious to the consumer that they need to explicitly do error handling. Compare to promises, where consumers may often use `.then` without remembering to handle rejections, which leads to unhandled promise rejections if not in an async function stack.

Downside is that style doesn't really play well with TypeScript typings:
https://github.com/derbyjs/racer/pull/305#discussion_r1612306992